src/config/read.js

```diff
@@ -66,6 +66,8 @@ function readRawConfig(argv, root) {
   }
 
   if (typeof argv.config === 'object') {
+    argv.config.name = argv.config.name || root.replace(/[/\\]|\s/g, '-');
+    argv.config.rootDir = argv.config.rootDir || root;
```

Comment:
Instead of adding it in `read.js` can we add it here: https://github.com/facebook/jest/blob/master/packages/jest-cli/src/config/loadFromFile.js#L27 ?